packages/firestore/src/remote/watch_change.ts

```diff
@@ -352,6 +352,24 @@ export class WatchChangeAggregator {
     });
   }
 
+  /**
+   * Iterates over all targetIds that the watch change applies to: either the
+   * targetIds explicitly listed in the change or the targetIds of all currently
+   * active targets.
+   */
+  forEachTargetId(
+    targetChange: WatchTargetChange,
+    fn: (targetId: TargetId) => void
+  ): void {
+    if (targetChange.targetIds.length > 0) {
+      targetChange.targetIds.forEach(fn);
+    } else {
+      objUtils.forEachNumber(this.targetStates, (targetId, _) => {
```

Comment:
This can be simplified to:

```
      objUtils.forEachNumber(this.targetStates, fn);
```
😛

Reply:
That loses the `this` context and always creates really hard to debug issues in the long run.

Reply:
While I think I know where you're coming from (doing `objUtils.forEachNumber(..., this.someFunction)` would be dangerous), I don't think it applies here since `fn` is just a free-standing function, not a member function... plus we're already doing the exact same construct 2 lines up.

That said, I don't care strongly in the least which option we do. I just wanted to nit your nit.

Reply:
Done.